File: String/Anagram/anagram.c
const int TRUE  = 1;
const int FALSE = 0;

int anagram(char* s, int sLen, const char* t, int tLen)
{
	if(s == null || t == null)
		return FALSE;
	
	if(sLen != tLen)
		return FALSE;
	
	int count[256] = {0};
	
	for(int i = 0; i < sLen; i++) {
		++count[s[i]];
		--count[t[i]];
	}
	
	for(int i = 0; i != tLen; i++) {
		if(count[t[i]] != 0)
			return FALSE;
	}
	return TRUE;
}

File: String/Anagram/Anagram.java
public class Anagram {
	public boolean anagram(String s, String t) {
		if(s == null || t == null)
			return false;
		
		if(s.length() != t.length())
			return false;
		
		final int NUM = 256;
		
		int[] count = new int[NUM];
		
		for(int i = 0; i < s.length(); i++) {
			++count[s[i]];
			--count[t[i]];
		}
		
		if(int i = 0; i != NUM; i++) {
			if(count[t[i]] != 0)
				return false;
		}
		return true;
	}
}